src/createTokenBridge.integration.test.ts

```diff
@@ -197,59 +198,33 @@ it(`successfully deploys token bridge contracts with a custom fee token through
   expect(fundTxReceipt.status).toEqual('success');
 
   // -----------------------------
-  // 2. check the new token bridge creator allowance
-  // NOTE: We will estimate the token bridge creation through the canonical TokenBridgeCreator
-  //       and we will later change the destination address to the new deployed TokenBridgeCreator.
-  //       Thus, we need to give allowance to both TokenBridgeCreators
-  const allowanceParams = {
+  // 2. approve custom fee token to be spent by the TokenBridgeCreator
+  const allowanceParams: CreateTokenBridgeEnoughCustomFeeTokenAllowanceParams = {
     nativeToken: testnodeInformation.l3NativeToken,
     owner: l3RollupOwner.address,
     publicClient: nitroTestnodeL2Client,
+    tokenBridgeCreatorAddressOverride: tokenBridgeCreator,
   };
 
-  // 2.a. Approval for canonical TokenBridgeCreator (gas estimation is still done through it, so we need allowance)
-  if (!(await createTokenBridgeEnoughCustomFeeTokenAllowance(allowanceParams))) {
```

Comment:
no longer necessary as gas estimation will now also be done with the provided factory